docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
 # Developer Guide
 
-## Acknowledgements
+## Table of Contents
+
+1. [Acknowledgements](#acknowledgements)
+2. [Design](#design)
+   - [Architecture](#architecture)
+   - [UI component](#ui-component)
+   - [Parser component](#parser-component)
+   - [Storage component](#storage-component)
+   - [Common classes](#common-classes)
```

Comment:
The link here for the common classes does not seem to be working, perhaps edit the content in the link bracket to be #common-class for it to work?